File: Server/src/main/java/com/augmentedcooking/Models/Rabbitmq/Response/complete/AiChatCompleteResponse.java
package com.augmentedcooking.Models.Rabbitmq.Response.complete;

import com.augmentedcooking.Enums.Response.ResponseTypes;
import com.augmentedcooking.Models.Rabbitmq.Response.AiChatResponse;
import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonTypeName;

import lombok.Getter;

import java.util.List;

@Getter
@JsonTypeName("COMPLETE")
public class AiChatCompleteResponse extends AiChatResponse {

    private final String role;
    private final String model;
    private final long timestamp;
    private final String userId;
    private final AiChatResponseMessage content;

    @JsonCreator
    public AiChatCompleteResponse(
            @JsonProperty("request_id") String requestId,
            @JsonProperty("user_id") String userId,
            @JsonProperty("type") String type,
            @JsonProperty("role") String role,
            @JsonProperty("model") String model,
            @JsonProperty("timestamp") long timestamp,
            @JsonProperty("content") AiChatResponseMessage content) {
        super(requestId, ResponseTypes.valueOf(type));
        this.userId = userId;
        this.role = role;
        this.model = model;
        this.timestamp = timestamp;
        this.content = content;
    }

    @Getter
    public static class AiChatResponseMessage {

        private final String text;
        private final List<AiChatResponseMessageImage> images;

        @JsonCreator
        public AiChatResponseMessage(
                @JsonProperty("text") String text,
                @JsonProperty("images") List<AiChatResponseMessageImage> images) {
            this.text = text;
            this.images = images;
        }

        @Getter
        public static class AiChatResponseMessageImage {

            private final String data;
            private final String format;
            private final int width;
            private final int height;
            private final String description;

            @JsonCreator
            public AiChatResponseMessageImage(
                    @JsonProperty("data") String data,
                    @JsonProperty("format") String format,
                    @JsonProperty("width") int width,
                    @JsonProperty("height") int height,
                    @JsonProperty("description") String description) {
                this.data = data;
                this.format = format;
                this.width = width;
                this.height = height;
                this.description = description;
            }
        }
    }
}

File: Clients/desktop/src/Models/Response/Chat/Complete/AiChatCompleteResponse.cs
using System.Text.Json.Serialization;

namespace AugmentedCooking.src.Models.Response.Chat.Complete;

public class AiChatCompleteResponse : AiChatResponse {
    [JsonPropertyName("user_id")]
    public required string UserId { get; set; }

    [JsonPropertyName("role")]
    public required string Role { get; set; }

    [JsonPropertyName("model")]
    public required string Model { get; set; }

    [JsonPropertyName("timestamp")]
    public required long Timestamp { get; set; }

    [JsonPropertyName("content")]
    public required AiChatResponseMessage Content { get; set; }
}

public class AiChatResponseMessage {
    [JsonPropertyName("text")]
    public required string Text { get; set; }

    [JsonPropertyName("images")]
    public required IList<AiChatResponseMessageImage> Images { get; set; }
}

public class AiChatResponseMessageImage {
    [JsonPropertyName("data")]
    public required string Data { get; set; }

    [JsonPropertyName("format")]
    public required string Format { get; set; }

    [JsonPropertyName("width")]
    public required int Width { get; set; }

    [JsonPropertyName("height")]
    public required int Height { get; set; }

    [JsonPropertyName("description")]
    public required string Description { get; set; }
}
